### PR TITLE
Textarea available without redux

### PR DESCRIPTION
**Pull request by pochretien**

### CODE REVIEW

File: src/components/textarea/tests/TextArea.spec.tsx [hunk 1]
```diff
@@ -77,6 +77,15 @@ describe('TextArea', () => {
...
 
+        it('should call prop onChangeCallBack on textarea change', () => {
+            const onChangeCallBack = jasmine.createSpy('onChangeCallBack');
+
+            wrapper.setProps({onChangeCallBack});
+            textArea.simulate('change');
+
+            expect(onChangeCallBack).toHaveBeenCalledTimes(1);
+        });
+
```
Comment (sroy3) [inline src/components/textarea/tests/TextArea.spec.tsx]: You'll need to add tests calling `handleOnChange` without `onChange` and `onChangeCallBack` and check that it does not throw or else the coverage will go down.

File: src/components/textarea/TextArea.tsx [hunk 2]
```diff
@@ -24,6 +24,8 @@ export interface ITextAreaOwnProps {
     disabledOnMount?: boolean;
 
     isAutosize?: boolean;
+
+    onChangeCallBack?: (event: React.KeyboardEvent<HTMLTextAreaElement>) => void;
```
Comment (sroy3) [inline src/components/textarea/TextArea.tsx]: Why not just use `onChange` if you aren't using Redux?

Comment (pochretien) [reply src/components/textarea/TextArea.tsx]: I added this function for the redux use case ( if you want to do something onChange without removing the redux behaviour.

File: src/components/textarea/tests/TextArea.spec.tsx (hunk 1 above)
Comment (pochretien) [reply src/components/textarea/tests/TextArea.spec.tsx]: I tried to do something for the onChange not defined but the not.throw want a function as argument. I check for an other solution to test.

File: src/components/textarea/TextArea.tsx [hunk 3]
```diff
@@ -80,10 +82,20 @@ export class TextArea extends React.Component<ITextAreaProps, {}> {
...
-                onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => this.props.onChange(e)}
+                onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => this.handleOnChange(e)}
             />
         );
     }
+
+    private handleOnChange(e: React.ChangeEvent<HTMLTextAreaElement>) {
+        if (this.props.onChange) {
+            this.props.onChange(e);
+        }
```
Comment (gdostie) [inline src/components/textarea/TextArea.tsx]: `callIfDefined(this.props.onChange, e);`

File: src/components/textarea/TextArea.tsx [hunk 4]
```diff
@@ -80,10 +82,20 @@ export class TextArea extends React.Component<ITextAreaProps, {}> {
...
+
+        if (this.props.onChangeCallBack) {
+            this.props.onChangeCallBack(e);
+        }
```
Comment (gdostie) [inline src/components/textarea/TextArea.tsx]: `callIfDefined(this.props.onChangeCallBack, e);`

File: src/components/textarea/TextArea.tsx (hunk 3 above)
Comment (pochretien) [reply src/components/textarea/TextArea.tsx]: oh nice!

**Pull request merged**